pkgs/by-name/co/containerd/package.nix

```diff
@@ -1,56 +1,82 @@
-{ lib
-, fetchFromGitHub
-, buildGoModule
-, btrfs-progs
-, go-md2man
-, installShellFiles
-, util-linux
-, nixosTests
-, kubernetes
+{
+  lib,
+  stdenv,
+  btrfs-progs,
+  fetchFromGitHub,
+  go,
+  go-md2man,
+  kubernetes,
+  nix-update-script,
+  nixosTests,
+  util-linux,
+  btrfsSupport ? btrfs-progs != null,
 }:
 
-buildGoModule rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "containerd";
-  version = "1.7.23";
+  version = "2.0.0";
+
+  outputs = [
+    "out"
+    "doc"
+    "man"
+  ];
 
   src = fetchFromGitHub {
     owner = "containerd";
     repo = "containerd";
-    rev = "v${version}";
-    hash = "sha256-vuOefU1cZr1pKCYHKyDBx/ohghgPlXhK3a38PQKH0pc=";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-DFAP+zjBYP2SpyD8KXGvI3i/PUZ6d4jdzGyFfr1lzj4=";
   };
 
-  vendorHash = null;
+  postPatch = "patchShebangs .";
+
+  strictDeps = true;
 
-  nativeBuildInputs = [ go-md2man installShellFiles util-linux ];
+  nativeBuildInputs = [
+    go
+    go-md2man
+    util-linux
+  ];
 
   buildInputs = [ btrfs-progs ];
 
-  BUILDTAGS = lib.optionals (btrfs-progs == null) [ "no_btrfs" ];
+  tags = lib.optional (!btrfsSupport) "no_btrfs";
 
-  buildPhase = ''
-    runHook preBuild
-    patchShebangs .
-    make binaries "VERSION=v${version}" "REVISION=${src.rev}"
-    runHook postBuild
-  '';
+  makeFlags = [
+    "PREFIX=${placeholder "out"}"
+
+    "BUILDTAGS=${toString finalAttrs.tags}"
+    "REVISION=${finalAttrs.src.rev}"
+    "VERSION=v${finalAttrs.version}"
+  ];
 
-  installPhase = ''
-    runHook preInstall
-    install -Dm555 bin/* -t $out/bin
-    installShellCompletion --bash contrib/autocomplete/ctr
-    installShellCompletion --zsh --name _ctr contrib/autocomplete/zsh_autocomplete
-    runHook postInstall
+  installTargets = "install install-doc install-man";
+
+  postConfigure = ''
+    export GOCACHE=$TMPDIR/go-cache
+    export GOPATH="$TMPDIR/go"
```

Comment:
I'm not 100% sure that `GOPATH` is needed here, but

I think this might be useful:
```nix
disallowedReferences = [ go ];
inherit (go) GOOS GOARCH;
```